### DISCUSSION TITLE
Upload Signed URL 403

### DISCUSSION THREAD

**Discussion by arnefromsa** [aws-solutions-library-samples/guidance-for-aws-sustainability-insights-framework#25] [General]
After creating a pipeline execution the signed URL returned give me a 403 error. I am using the following to push the file to the URL : 

```

        file=open(localFilepath,"r")
        payload = file.read()
        headers = {
        'Content-Type': 'text/csv'
        }

        response = requests.request("POST", url, headers=headers, data=payload)
```

The response I'm getting for the URL (inputUploadUrl) is a <Response [403]> for the request.  Should I be using my Cognito Auth Token as well puthing to the S3 url ?

**Comment by johnrotach** [comment]
Hello @arnefromsa, can you try the same request as a `PUT`? Through cURL I was able to validate that if I do a `PUT` request the request succeeds and if I change that same request to a `POST` I get a `403 - Forbidden - SignatureDoesNotMatch` error. Reply back if this doesn't fix your issue and we can try to get you a requests sample.

To answer your specific question you do not need to provide the auth token with this request as auth is taken care of by the signature contained in the URL.

**Reply by arnefromsa** [reply]
Thanks @johnrotach - using PUT worked, such a newbie error.  

I understood that the AUTH will not be needed as the pattern used is built for the file upload to be via the signature.
